src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -336,11 +359,66 @@ public void CopyChangedValues(TStructuralType original)
                     bool isDeltaType = TypedDelta.IsDeltaOfT(deltaNestedResource.GetType());
```

Comment:
>bool isDeltaType = TypedDelta.IsDeltaOfT(deltaNestedResource.GetType()); [](http://example.com/codeflow?start=20&length=72)

I realize this isn't your change, but we should put this in the Contract.Assert so that we're not making this somewhat expensive call unnecessarily in the non-debug case.

Reply:
Since deltaNestedResource is a dynamic type it doenst allow to use inside contract.assert. throws a build err


Error CS1974 The dynamically dispatched call to method 'Assert' may fail at runtime because one or more applicable overloads are conditional methods.

Reply:
That's a bummer.  

At least we should be able to put the TypedDelta.IsDeltaOfT() call within the Contract.Assert:

Type deltaType = deltaNestedResource.GetType();
Contract.Assert(TypedDelta.IsDeltaOfT(deltaType),...);

Reply:
Alternatively, we could cast to object:
 Contract.Assert(TypedDelta.IsDeltaOfT(((object)deltaNestedResource)?.GetType()), nestedResourceName + "'s corresponding value should be Delta<T> type but is not.");